components/ws-proxy/cmd/root.go

```diff
@@ -114,3 +150,30 @@ func getConfig(fn string) (*Config, error) {
 
 	return &cfg, nil
 }
+
+// Duration is a json-unmarshallable wrapper for time.Duration.
+// See https://stackoverflow.com/questions/48050945/how-to-unmarshal-json-into-durations
+type Duration time.Duration
+
+// UnmarshalJSON unmarshales a duration using ParseDuration
+func (d *Duration) UnmarshalJSON(b []byte) error {
+	var v interface{}
+	if err := json.Unmarshal(b, &v); err != nil {
+		return err
+	}
+
+	switch value := v.(type) {
+	case float64:
+		*d = Duration(time.Duration(value))
+		return nil
+	case string:
+		tmp, err := time.ParseDuration(value)
+		if err != nil {
+			return err
+		}
+		*d = Duration(tmp)
+		return nil
+	default:
+		return xerrors.New("invalid duration")
+	}
+}
```

Comment:
this already exists in https://github.com/gitpod-io/gitpod/blob/b7dda84ad0c0dd195b829b7f4160f9f7acfa355c/components/common-go/util/duration.go#L14-L48

We should be able to re-use that bit, rather than copy.